gcloud-java-storage/src/main/java/com/google/gcloud/spi/DefaultStorageRpc.java

```diff
@@ -368,7 +368,11 @@ public void onSuccess(Void ignore, HttpHeaders responseHeaders) {
 
           @Override
           public void onFailure(GoogleJsonError e, HttpHeaders responseHeaders) {
-            deletes.put(tuple.x(), Tuple.<Boolean, StorageException>of(null, translate(e)));
+            if (e.getCode() == 404) {
```

Comment:
Sorry, my mess but I wonder if we can clean it in this PR or create another one for it:
We deal with 404 in both places `DefaultStorageRpc` for deletes and in `StorageImpl` for gets.
I think it is fine (and probably simpler) to move all of it to `DefaultStorageRpc` and it would be nice 
to mention the "return null" case in `StorageRpc`.

Also, we could use `java.net.HttpURLConnection.HTTP_NOT_FOUND` instead of the numerical value `404`.